Source/Classes/TextMask.lua

```diff
@@ -105,10 +109,36 @@ function TextMaskClass:SetMaskType(name)
 	end
 	self._MaskType = mask
 	self.Frame.Text = mask:Process(self.Frame.Text):sub(1, self._MaxLength)
+	
+	-- Add increment/decrement buttons for numeric inputs
+	if name == "Number" then
+		-- Initialize text to 0, consumer can override
+		self.Frame.Text = 0
+
+		self.incrementButton = INCREMENT_BUTTON:Clone()
+		self.incrementButton.Position = UDim2.new(0, self.Frame.AbsoluteSize.X - self.Frame.AbsoluteSize.Y / 2, 0, 0)
+		self.incrementButton.Activated:Connect(function()
+			self.Frame.Text = tonumber(self.Frame.Text) + 1
+		end)
+		self.incrementButton.Parent = self.Frame
+		
+		self.decrementButton = DECREMENT_BUTTON:Clone()
+		self.decrementButton.Position = UDim2.new(0, self.Frame.AbsoluteSize.X - self.Frame.AbsoluteSize.Y / 2, 0, self.Frame.AbsoluteSize.Y / 2)
```

Comment:
ok i can't trust absolute size so i'm going to figure out a better way to do this

Reply:
Similaliry to the comment above:

```suggestion
self.DecrementButton.AnchorPoint = Vector2.new(1, 1)
self.DecrementButton.Position = UDim2.new(1, 0, 1, 0)
```